cowprotocol/cowswap
# feat(confirmation-modal): ✨ Add confirmation modal with relevant hooks

# Summary
When our application is loaded in an iframe, we have UX breaking issues whenever we'd like to confirm with the user if they'd like to take certain actions, such as removing a list.

This happens since we are relying on `window.prompt` which is a sandboxed feature. So in places such as Safe app we end up with a broken experience.

This PR adds a new feature: `useConfirmationRequest`. This hook can be used anywhere, to request for the user to confirm whether or not they'd like to confirm what they are doing.

As an example, I've refactored the **Remove List** check which normally looks as following;

![Screenshot 2023-04-24 at 17 56 28](https://user-images.githubusercontent.com/5172699/234051161-0b1630fc-13cf-4424-b9f0-e4bd2db88463.png)

After these changes, it functions as the following;

![Screenshot 2023-04-24 at 17 53 21](https://user-images.githubusercontent.com/5172699/234051241-5ad9b361-986d-4dc4-a0bc-64802748d8df.png)

# To Test

1. Go to application
2. Click on a currency
3. Click on manage token lists
4. Try to remove one via the gear icon, and click on remove list
5. You should see the new confirmation modal

# Background
https://cowservices.slack.com/archives/C0361CDG8GP/p1682078523149949

# Follow up
Once approach is agreed upon, clearing out all the `window.prompt` calls which should be easy.
